develop_tag.rb

```diff
@@ -0,0 +1,135 @@
+class DevelopTag
+
+  def update_develop_tag
+    if !list_all_tags.empty?
+      assign_new_dev_tag
+    else
+      assign_first_dev_tag
+    end
+  end
+
+  def list_all_tags
+    list_each_tag = `git tag`
+    list_each_tag = list_each_tag.split("\n")
+    return list_each_tag
+  end
+
+  def list_tag_environ_and_version_num
+    list_each_tag = list_all_tags
+    i=0
+    tag_id_list = []
```

Comment:
instead of creating an array and pushing to it you could achieve the while loop below with a 
```
.select { #starts with dev}
.map { #map to the version}
```